### ISSUE TITLE
Exit code propagation of exceptions

### ISSUE THREAD

**Issue by pieterlukasse** [opentargets/platform#295]
A FATAL error (corrupt .json.gz file) was not reported as fatal and the --val step ended with "INFO" log level message stating it "finished". See log lines below.

Steps to reproduce:
- download one of the .json.gz files (chembl in my case) and make sure it gets corrupted (i.e. edit out some lines manually)
- run the --val step

```
2018-11-07 14:29:30,635 - mrtarget.modules.EvidenceValidation - INFO - generate_validator_from_schema literature using the uri https://raw.githubusercontent.com/opentargets/json_schema/1.2.8/src/literature_mining.json
2018-11-07 14:29:30,651 - mrtarget.modules.EvidenceValidation - INFO - generate_validator_from_schema genetic_literature using the uri https://raw.githubusercontent.com/opentargets/json_schema/1.2.8/src/literature_curated.json
2018-11-07 14:29:30,671 - mrtarget.modules.EvidenceValidation - INFO - generate_validator_from_schema literature using the uri https://raw.githubusercontent.com/opentargets/json_schema/1.2.8/src/literature_mining.json
2018-11-07 14:29:34,869 - mrtarget.modules.EvidenceValidation - ERROR - Error processing job 571957d261b941f588517e3fa9bcf7fb: CRC check failed 0xcf529d48 != 0xeb43362L
Traceback (most recent call last):
  File "mrtarget/common/Redis.py", line 669, in _inner_run
    job_results = self.process(data)
  File "mrtarget/modules/EvidenceValidation.py", line 214, in process
    md5_hash, logfile=logfile)
  File "mrtarget/modules/EvidenceValidation.py", line 225, in parse_file
    n_lines = self._count_file_lines(f_handler)
  File "mrtarget/modules/EvidenceValidation.py", line 252, in _count_file_lines
    return sum(1 for el in file_handle)
  File "mrtarget/modules/EvidenceValidation.py", line 252, in <genexpr>
    return sum(1 for el in file_handle)
  File "/usr/local/lib/python2.7/gzip.py", line 464, in readline
    c = self.read(readsize)
  File "/usr/local/lib/python2.7/gzip.py", line 268, in read
    self._read(readsize)
  File "/usr/local/lib/python2.7/gzip.py", line 315, in _read
    self._read_eof()
  File "/usr/local/lib/python2.7/gzip.py", line 354, in _read_eof
    hex(self.crc)))
IOError: CRC check failed 0xcf529d48 != 0xeb43362L
2018-11-07 14:29:34,870 - mrtarget.modules.EvidenceValidation - INFO - FileReaderProcess-2 done processing
2018-11-07 14:29:34,968 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-34 done processing
2018-11-07 14:29:34,968 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-45 done processing
...
2018-11-07 14:29:35,868 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-35 done processing
2018-11-07 14:29:35,868 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-29 done processing
2018-11-07 14:29:35,868 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-38 done processing
2018-11-07 14:29:35,868 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-10 done processing
2018-11-07 14:29:35,969 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-31 done processing
2018-11-07 14:29:35,969 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-37 done processing
2018-11-07 14:29:35,969 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-40 done processing
2018-11-07 14:29:35,969 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-36 done processing
2018-11-07 14:29:35,969 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-39 done processing
2018-11-07 14:29:36,069 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-33 done processing
2018-11-07 14:29:36,069 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-27 done processing
2018-11-07 14:29:36,069 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-26 done processing
2018-11-07 14:29:36,069 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-30 done processing
2018-11-07 14:29:36,069 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-19 done processing
2018-11-07 14:29:36,169 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-50 done processing
2018-11-07 14:29:36,169 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-3 done processing
2018-11-07 14:29:36,169 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-46 done processing
2018-11-07 14:29:36,169 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-44 done processing
2018-11-07 14:29:36,169 - mrtarget.modules.EvidenceValidation - INFO - ValidatorProcess-32 done processing
2018-11-07 14:29:36,196 - mrtarget.modules.EvidenceValidation.EvidenceValidationFileChecker - INFO - collecting readers
2018-11-07 14:29:36,212 - mrtarget.common.ElasticsearchLoader - WARNING - optimisation of index 18.10_validated-data-chembl failed
2018-11-07 14:29:36,212 - mrtarget.modules.EvidenceValidation.EvidenceValidationFileChecker - INFO - collecting reporter
2018-11-07 14:29:56,876 - mrtarget.modules.EvidenceValidation.EvidenceValidationFileChecker - INFO - QC on prefix chembl
2018-11-07 14:29:56,928 - __main__.main() - INFO - `/usr/src/app/mrtarget/CommandLine.py 18.10 --val --log-level INFO --schema-version 1.2.8 --input-file /tmp//ot_files/downloads/18.10/chembl-27-09-2018.json.gz` - finished
~
```

Desired behavior:
- this error should be reported as FATAL.
- the --val step should finish in ERROR state. Last log message should be a ERROR level message stating that the step failed and could *not* finish. 
- the --val step should end with exit code != 0

**Comment by afaulconbridge** [comment]
This is a more general issue than just the `--val` step. We need to ensure that an appropriate exit code (i.e. not zero) is returned by the data pipeline form any stage and for any exception. I've updated the title and tags of this ticket to reflect that.

**Comment by afaulconbridge** [comment]
This seems to be an issue with the Makefile rather than the pipeline. opentargets/data_pipeline#380 should resolve

**Comment by afaulconbridge** [comment]
Closing this ticket opentargets/data_pipeline#380 fixes the issue, and will be included in 18.12.0 release shortly.

**Comment by pieterlukasse** [comment]
@afaulconbridge I wasn't running the makefile.
